### PR TITLE
SBA-27 Created restaurant table

### PR DESCRIPTION
**Pull request by Shazil-55**

### CODE REVIEW

File: src/database/migrations/20231122000019_create_restaurantDetails.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Knex } from 'knex';
+
+function up(knex: Knex) {
+  return knex.schema.createTable('RestaurantDetails', (t) => {
+    t.uuid('id').unique().defaultTo(knex.raw('gen_random_uuid()')).primary();
+    t.uuid('restaurantId').notNullable();
+    t.foreign('restaurantId').references('id').inTable('Restaurant');
```
Comment (asim47) [inline src/database/migrations/20231122000019_create_restaurantDetails.ts:7]: Why are there two restaurantId?

File: src/database/migrations/20231122000018_create_restaurant.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Knex } from 'knex';
+
+function up(knex: Knex) {
+  return knex.schema.createTable('Restaurant', (t) => {
```
Comment (asim47) [inline src/database/migrations/20231122000018_create_restaurant.ts:4]: r small

File: src/database/migrations/20231122000019_create_restaurantDetails.ts (hunk 1 above)
Comment (asim47) [inline src/database/migrations/20231122000019_create_restaurantDetails.ts:7]: Restaurant r small

File: src/database/migrations/20231122000019_create_restaurantDetails.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { Knex } from 'knex';
+
+function up(knex: Knex) {
+  return knex.schema.createTable('RestaurantDetails', (t) => {
```
Comment (asim47) [inline src/database/migrations/20231122000019_create_restaurantDetails.ts:4]: r small

File: src/database/migrations/20231122000019_create_restaurantDetails.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    t.specificType('paymentMethod', 'text[]');
+    t.specificType('allergies', 'text[]');
+    t.boolean('isVegan').notNullable();
+    t.boolean('isVegetarian').notNullable();
+    t.boolean('allowSelfPickup').notNullable();
+    t.boolean('allowTableBook').notNullable();
+    t.boolean('allowHomeDelivery').notNullable();
+    t.string('generalInformation').notNullable();
+    t.string('attributes').notNullable();
+    t.string('specialities').notNullable();
```
Comment (asim47) [inline src/database/migrations/20231122000019_create_restaurantDetails.ts:21]: t.string('generalInformation').notNullable();
    t.string('attributes').notNullable();
    t.string('specialities').notNullable();
    
    can be nullAble

**Review by asim47 — COMMENTED** [review]
Also add entities.

include the restaurantDetails? ResturantDetails in Restaurant entity

**Review by asim47 — CHANGES_REQUESTED** [review]
Please resolve all the comments

File: src/database/migrations/20231122000019_create_restaurantDetails.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+function up(knex: Knex) {
+  return knex.schema.createTable('RestaurantDetails', (t) => {
+    t.uuid('id').unique().defaultTo(knex.raw('gen_random_uuid()')).primary();
+    t.uuid('restaurantId').notNullable();
+    t.foreign('restaurantId').references('id').inTable('Restaurant');
+    t.string('phoneNo').notNullable();
+    t.string('email').notNullable();
+    t.boolean('acceptCoupons').notNullable();
+    t.boolean('freeShipping').notNullable();
+    t.specificType('paymentMethod', 'text[]');
```
Comment (asim47) [inline src/database/migrations/20231122000019_create_restaurantDetails.ts:12]: Three different checks for payment methods

**Review by asim47 — CHANGES_REQUESTED** [review]

File: src/database/migrations/20231122000019_create_restaurantDetails.ts (hunk 1 above)
Comment (Shazil-55) [reply src/database/migrations/20231122000019_create_restaurantDetails.ts:7]: Line 6 is the column 
and line 7 is refrancing the restauratnt id to id in restaurant table

File: src/helpers/entities.ts [hunk 6]
```diff
@@ -79,3 +79,44 @@ export interface VerifyOtpForAccountRecovery {
...
+  deliveryAmount: number;
+  averageRating: number;
+  vendorId: string;
+  restaurantDetails?: RestaurantDetails;
+  createdAt: string;
+  updatedAt: string;
+}
+
+export interface RestaurantDetails {
+  acceptCoupons: boolean;
```
Comment (asim47) [inline src/helpers/entities.ts:102]: id missing

Comment (asim47) [reply src/helpers/entities.ts:102]: Restaurant id also missing

**Review by asim47 — CHANGES_REQUESTED** [review]
Table and entities dont match

**Review by asim47 — APPROVED** [review]

**Pull request merged**